paginator/paginator.go

```diff
@@ -163,6 +163,14 @@ func isNil(i interface{}) bool {
 	if i == nil {
 		return true
 	}
+	// for custom types, evaluate isNil on the underlying value
+	if ct, ok := i.(cursor.CustomType); ok {
+		val, err := ct.GetCustomTypeValue(i)
```

Comment:
`GetCustomTypeValue` should receive special metadata as its argument, taking `JSON` custom type as example, it should be the field name:

https://github.com/pilagod/gorm-cursor-paginator/blob/f807e66213b074b491dff8cf5e7fc3040f9768df/paginator/paginator_test.go#L73

I am wondering that this line uses custom object itself as the metadata would cause problem in some cases 🤔

Reply:
Maybe it would be better to handle custom type in `decodeCursor`. In that function, we can access metadata from `p.rules` and call `GetCustomTypeValue` properly.

Reply:
You're absolutely correct. I guess I was just focusing on my case where I didn't use metadata :see_no_evil:
Thanks!
Moving the logic to `decodeCursor`.